code/modules/food_and_drinks/kitchen_machinery/gibber.dm

```diff
@@ -135,6 +135,31 @@
 	dump_inventory_contents()
 	update_appearance()
 
+/obj/machinery/gibber/proc/start_shaking()
+	var/static/list/transforms
+	if(!transforms)
+		var/matrix/M1 = matrix()
+		var/matrix/M2 = matrix()
+		var/matrix/M3 = matrix()
+		var/matrix/M4 = matrix()
+		M1.Translate(-1, 0)
+		M2.Translate(0, 1)
+		M3.Translate(1, 0)
+		M4.Translate(0, -1)
+		transforms = list(M1, M2, M3, M4)
+	animate(src, transform=transforms[1], time=0.4, loop=-1)
+	animate(transform=transforms[2], time=0.2)
+	animate(transform=transforms[3], time=0.4)
+	animate(transform=transforms[4], time=0.6)
```

Comment:
Also I feel like you could put the `.Translate()`'s on the same line as they're initialized, but I didn't suggest it because I'm not entirely sure how much better that would be.

That being said, does a time of `0.4` actually work? I thought DM only worked up to deciseconds, which it uses as a base for time measurements as an integer. (Basically this means that all of these time values would likely be considered as `1` by the engine, I think).

This suggestion applies to all the three other places where this is copy-pasted. That being said, I feel like this could almost be a component or something, instead, since the same code is basically copy-pasted multiple times.
```suggestion
	if(!transforms)
		var/matrix/first_translation = matrix()
		var/matrix/second_translation = matrix()
		var/matrix/third_translation = matrix()
		var/matrix/fourth_translation = matrix()
		first_translation.Translate(-1, 0)
		second_translation.Translate(0, 1)
		third_translation.Translate(1, 0)
		fourth_translation.Translate(0, -1)
		transforms = list(first_translation, second_translation, third_translation, fourth_translation)

	animate(src, transform = transforms[1], time = 0.4, loop=-1)
	animate(transform = transforms[2], time = 0.2)
	animate(transform = transforms[3], time = 0.4)
	animate(transform = transforms[4], time = 0.6)
```

Reply:
https://github.com/tgstation/tgstation/pull/68885/commits/e41d4d9a96e71d175d428c68cd9f9af670e8a721